tfsdk/serve.go

```diff
@@ -208,12 +220,24 @@ func (s *server) GetProviderSchema(ctx context.Context, _ *tfprotov6.GetProvider
 	resp.ProviderMeta = providerMeta6Schema
 	resp.ResourceSchemas = resource6Schemas
 	resp.DataSourceSchemas = dataSource6Schemas
-	return resp, nil
+}
+
+// validateProviderConfigResponse is a thin abstraction to allow native Diagnostics usage
```

Comment:
nitpick: ideally we'd call out _why_ this allows native Diagnostics usage, but we can fast-follow with that.